beetsplug/alternatives.py

```diff
@@ -125,19 +126,24 @@ def parse_config(self, config):
 
     def matched_item_action(self, item):
         path = self.get_path(item)
+        actions = []
         if path and os.path.isfile(path):
             dest = self.destination(item)
             if path != dest:
-                return (item, self.MOVE)
+                actions.extend([self.MOVE, self.WRITE])
```

Comment:
I guess the `self.WRITE` action is here so that we cover the case when a file has been moved _and_ updated, correct?

Couldn’t we achieve the same thing with this code?
~~~python
if path != dest:
    actions.apped(self.MOVE)
if (os.path.getmtime(syspath(dest))
        < os.path.getmtime(syspath(item.path))):
    actions.append(self.WRITE)
~~~

Reply:
I guess so. However, the write operation in that situation is not new actually, I only moved it from [the `update()` function](https://github.com/geigerzaehler/beets-alternatives/pull/17/files/94742e403a049d80571e9746f26c6bc0c5c41523#diff-ebb574ba2f6ca42686b2991565c7dbe2L183) to `matched_item_action()`. I guess that was to disentangle the `MOVE` and `WRITE` operation (it is quite some time ago that I initially made this commit...) and certainly unrelated to the title of this PR.

To make your code suggestion work, it should check the file at `path` instead of the one at `dest`, though:
```python
if path != dest:
    actions.append(self.MOVE)
if (os.path.getmtime(syspath(path))
        < os.path.getmtime(syspath(item.path))):
    actions.append(self.WRITE)
```

That is probably more concise than the current code. There might be some (not so clear) implication on performance, although I think the limiting factor is likely to beets database performance anyway:
* if the `MOVE` is caused by a change to the path formats (i.e. the templates), your version could prevent unnecessary `WRITE`s since it checks mtime,
* if the `MOVE` is triggered by changed metadata (title, ... , whatever is used by the template), the `WRITE` is required and the original code saved on filesystem accesses by not calling `getmtime`.

I think I'm going to use the code you suggested.

Reply:
You’re right about `path` instead of `dest` and spot on with regard to the performance trade-off.